service/src/test/java/bio/terra/cbas/controllers/TestRunSetsApiController.java

```diff
@@ -0,0 +1,135 @@
+package bio.terra.cbas.controllers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+import bio.terra.cbas.dao.MethodDao;
+import bio.terra.cbas.dao.RunDao;
+import bio.terra.cbas.dao.RunSetDao;
+import bio.terra.cbas.model.ParameterDefinition;
+import bio.terra.cbas.model.ParameterDefinitionEntityLookup;
+import bio.terra.cbas.model.ParameterDefinitionLiteralValue;
+import bio.terra.cbas.model.ParameterValueType;
+import bio.terra.cbas.model.RunSetRequest;
+import bio.terra.cbas.model.RunSetStateResponse;
+import bio.terra.cbas.model.WdsEntitySet;
+import bio.terra.cbas.model.WorkflowParamDefinition;
+import bio.terra.cbas.models.Method;
+import bio.terra.cbas.models.Run;
+import bio.terra.cbas.models.RunSet;
+import cromwell.client.api.WorkflowsApi;
+import cromwell.client.model.WorkflowIdAndStatus;
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.UUID;
+import org.databiosphere.workspacedata.api.EntitiesApi;
+import org.databiosphere.workspacedata.model.EntityAttributes;
+import org.databiosphere.workspacedata.model.EntityResponse;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.springframework.http.ResponseEntity;
+
+public class TestRunSetsApiController {
+
+  final String wdsApiV = "v2";
+  final String wdsInstanceId = UUID.randomUUID().toString();
+
+  @Test
+  void chainCallsTogether() throws Exception {
```

Comment:
The intent was to reduce the tested scope only to the data flow through this one class... is there leakage somewhere?

I guess I'd be curious to hear what other options there would be for testing this (since you flagged the interfaces and dependency injection patterns too... maybe I just set about the whole problem the wrong way and there's an easier option...?)